node/test/Test/Pos/Ssc/GodTossing/Toss/PureSpec.hs

```diff
@@ -0,0 +1,100 @@
+module Test.Pos.Ssc.GodTossing.Toss.PureSpec
+       ( spec
+       ) where
+
+import           Universum
+
+import qualified Crypto.Random                     as Rand
+import           Data.Default                      (def)
+import           Test.Hspec                        (Spec, describe)
+import           Test.Hspec.QuickCheck             (modifyMaxSuccess, prop)
+import           Test.QuickCheck                   (Arbitrary (..), Gen, Property, forAll,
+                                                    (===))
+import           Test.QuickCheck.Arbitrary.Generic (genericArbitrary, genericShrink)
+
+import           Pos.Core                          (EpochOrSlot, HasCoreConstants,
+                                                    StakeholderId, addressHash,
+                                                    giveStaticConsts)
+import qualified Pos.Ssc.GodTossing                as Toss
+
+spec :: Spec
+spec = giveStaticConsts $ describe "Toss" $ do
+    let smaller n = modifyMaxSuccess (const n)
+    describe "PureToss" $ smaller 30 $ do
+        prop "Adding and deleting a signed commitment in the 'PureToss' monad is the\
+             \ same as doing nothing"
+             putDelCommitment
+        prop "Adding and deleting an opening in the 'PureToss' monad is the same as doing\
+             \ nothing"
+             putDelOpening
+        prop "Adding and deleting a share in the 'PureToss' monad is the same as doing\
+             \ nothing"
+             putDelShare
+
+data TossAction
+    = PutCommitment Toss.SignedCommitment
+    | PutOpening StakeholderId Toss.Opening
+    | PutShares StakeholderId Toss.InnerSharesMap
+    | PutCertificate Toss.VssCertificate
+    | ResetCO
+    | ResetShares
+    | DelCommitment StakeholderId
+    | DelOpening StakeholderId
+    | DelShares StakeholderId
+    | SetEpochOrSlot EpochOrSlot
+    deriving (Show, Eq, Generic)
+
+instance HasCoreConstants => Arbitrary TossAction where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+actionToMonad :: Toss.MonadToss m => TossAction -> m ()
+actionToMonad (PutCommitment sc)   = Toss.putCommitment sc
+actionToMonad (PutOpening sid o)   = Toss.putOpening sid o
+actionToMonad (PutShares sid ism)  = Toss.putShares sid ism
+actionToMonad (PutCertificate v)  = Toss.putCertificate v
+actionToMonad ResetCO              = Toss.resetCO
+actionToMonad ResetShares          = Toss.resetShares
+actionToMonad (DelCommitment sid)  = Toss.delCommitment sid
+actionToMonad (DelOpening sid)     = Toss.delOpening sid
+actionToMonad (DelShares sid)      = Toss.delShares sid
+actionToMonad (SetEpochOrSlot eos) = Toss.setEpochOrSlot eos
+
+emptyTossSt :: Toss.GtGlobalState
+emptyTossSt = def
+
+perform :: HasCoreConstants => [TossAction] -> Toss.PureToss ()
+perform = foldl (>>) (return ()) . map actionToMonad
+
+type TossTestInfo = (Word64, Word64, Word64, Word64, Word64)
+
+(==^) :: HasCoreConstants => [TossAction] -> [TossAction] -> TossTestInfo -> Property
```

Comment:
Write a haddock so that people would be able to understand what it does without having to look at the code.

Reply:
I was going to refactor this and add a typeclass to unify these tests and `PurePoll` ones, and as such the comments would be the same, but seeing as it might take a while until that happens, I added the comments you mentioned.